src/InterventionBackend.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+namespace SilverStripe\Assets;
+
+use Intervention\Image\Exception\NotWritableException;
+use SilverStripe\Assets\Storage\AssetContainer;
+use SilverStripe\Assets\Storage\AssetStore;
+use Intervention\Image\ImageManager as InterventionImage;
+
+class InterventionBackend implements Image_Backend
+{
+
+    /**
+     * @var AssetContainer
+     */
+    private $container;
+
+    /**
+     * @var \Intervention\Image\Image
+     */
+    private $image;
+
+    /**
+     * @var int
+     */
+    private $quality;
+
+    /**
+     * @var InterventionImage
+     */
+    private $manager;
+
+    public function __construct(AssetContainer $assetContainer = null)
+    {
+        $this->setAssetContainer($assetContainer);
+    }
+
+    public function getAssetContainer()
+    {
+        return $this->container;
+    }
+
+    public function setAssetContainer($assetContainer)
+    {
+        $this->container = $assetContainer;
+        return $this;
+    }
+
+    public function getImageManager()
+    {
+        if (!$this->manager) {
+            $this->setImageManager(new InterventionImage([ 'driver' => 'gd' ]));
+        }
+        return $this->manager;
+    }
+
+    public function setImageManager($manager)
+    {
+        $this->manager = $manager;
+        return $this;
+    }
+
+    public function loadFromContainer(AssetContainer $assetContainer)
+    {
+        $this->setImageResource($this->getImageManager()->make($assetContainer->getStream()));
+        return $this;
+    }
+
+    public function loadFrom($path)
+    {
+        $this->setImageResource($this->getImageManager()->make($path));
+        return $this;
+    }
+
+    public function getImageResource()
+    {
+        if (!$this->image && $this->getAssetContainer()) {
+            $this->loadFromContainer($this->getAssetContainer());
+        }
+        return $this->image;
+    }
+
+    /**
+     * @param \Intervention\Image\Image $image
+     * @return $this
+     */
+    public function setImageResource($image)
+    {
+        // make sure the image is correctly orientated
+        $image->orientate();
+        $this->image = $image;
+        return $this;
+    }
+
+    public function writeToStore(AssetStore $assetStore, $filename, $hash = null, $variant = null, $config = array())
+    {
+        // Write to temporary file, taking care to maintain the extension
+        $path = tempnam(sys_get_temp_dir(), 'intervention');
```

Comment:
I wanted to announce a small warning since I've worked with image manipulations in PHP, be careful of what image data that goes through PHP memory, especially larger files.